docs/authentication/jwt-auth.html

```diff
@@ -16,13 +16,407 @@
+        let authHeader = request.headers["Authorization"]
+            guard let authComponents = authHeader?.components(separatedBy: " "),
+                authComponents.count == 2,
+                authComponents[0] == "Bearer",
+                let jwt = try? JWT&lt;ClaimsStandardJWT&gt;(jwtString: authComponents[1], verifier: App.jwtVerifier)
+                else {
+                    let _ = response.send(status: .unauthorized)
+                    return try response.end()
+            }
+        response.send(jwt)
+        next()
+    }
+}
+
+extension App {
+    // Example for HMAC signer and verifier
+    static let jwtSigner = JWTSigner.hs256(key: Data("kitura".utf8))
+    static let jwtVerifier = JWTVerifier.hs256(key: Data("kitura".utf8))
+}</code></pre>
+    <div class="underline"></div>
+    <h2 class="heading-2"><span class="blue-text">Step 9:</span> JWTs on Codable Routes (Optional)</h2>
+    <p class="block-text">In our example we used raw routing since we wanted to access the request headers.
+         If we want to use JWTs on our codable routes, we need to encapulate this code in a TypeSafeMiddleware.</p>
+    <h2 class="heading-2"><span class="blue-text">Step 9a:</span> Define our TypeSafeJWT middleware.</h2>
+    <p>Create a new file called `TypeSafeJWT.swift` in `Sources` > `Application` > `Middlewares`</p>
+    <p>Inside this file, define `TypeSafeJWT` with the following code:</p>
+<pre><code class="language-swift">import SwiftJWT
+import Kitura
+
+struct TypeSafeJWT&lt;C: Claims&gt;: TypeSafeMiddleware {
+    static func handle(request: RouterRequest, response: RouterResponse, completion: @escaping (TypeSafeJWT<C>?, RequestError?) -> Void) {
+
+    }
+}</code></pre>
+    <p class="block-text">The `handle` function is required for a TypeSafeMiddleware and is where we will interact with the headers.
+    We will the return an instance of TypeSafeJWT on success or a RequestError on failure. The struct is generic so it can be used on any JWT.</p>
+    <p>Within the handler we are interested in the decoded JWT so we add that as a field:</p>
+    <pre><code class="language-swift">let jwt: JWT<C></code></pre>
+    <p>Finally, we initialize our JWT using the same functions as our protected get route.</p>
+    <p>Inside our handle function, add the following:</p>
+    <pre><code class="language-swift">let authHeader = request.headers["Authorization"]
+guard let authComponents = authHeader?.components(separatedBy: " "),
+    authComponents.count == 2,
+    authComponents[0] == "Bearer",
+    let jwt = try? JWT&lt;C&gt;(jwtString: authComponents[1], verifier: App.jwtVerifier)
+else {
+    return completion(nil, .unauthorized)
+}
+completion(TypeSafeJWT(jwt: jwt), nil)</code></pre>
+<p>Your completed `TypeSafeJWT.swift` file should look as follows:</p>
+<pre><code class="language-swift">import SwiftJWT
+import Kitura
+
+struct TypeSafeJWT&lt;C: Claims&gt;: TypeSafeMiddleware {
+    let jwt: JWT&lt;C&gt;
```

Comment:
Why do we define the JWT above as `let jwt: JWT` and then in here it's `let jwt: JWT<C>`?

Reply:
in the code we had let `jwt: JWT<C>` instead of `jwt: JWT&lt;C&gt;` so it wasn't showing up